src/Illuminate/Support/Stringable.php

```diff
@@ -235,6 +235,37 @@ public function convertCase(int $mode = MB_CASE_FOLD, ?string $encoding = 'UTF-8
         return new static(Str::convertCase($this->value, $mode, $encoding));
     }
 
+    /**
+     * Compresses a string using gzip compression.
+     *
+     * @param  int  $mode  Compression level (1 to 9, defaults to 5).
+     * @return string|false The compressed string, or false on failure.
+     */
+    public function compress(int $mode = 5)
+    {
+        if ($mode < 1 || $mode > 9) {
+            throw new \OutOfBoundsException('Compression mode must be between 1 and 9, default 5.');
+        }
+
+        return gzcompress($this->value, $mode);
```

Comment:
```suggestion
       return new static(Str::compress($this->value, $mode));
```

Reply:
It seems like an excellent way to do it, so we handle exceptions in a single method